junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -192,10 +192,44 @@ private FileVisitResult deleteAndContinue(Path path) {
 						Files.delete(path);
 					}
 					catch (IOException ex) {
-						failures.put(path, ex);
+						Throwable err = deleteUndeletableAndContinue(path, ex);
+						if (err != null) {
+							failures.put(path, ex);
+						}
 					}
 					return CONTINUE;
 				}
+
+				private Throwable deleteUndeletableAndContinue(Path path, IOException ex) {
+					boolean isWritable;
+					try {
+						isWritable = path.toFile().setWritable(true);
+					}
+					catch (Exception e) {
+						// In case setWritable throws we can't proceed
+						ex.addSuppressed(e);
+						return ex;
+					}
+					if (!isWritable) {
+						ex.addSuppressed(new Exception("Attempt to make file '" + path + "' writable failed") {
+							@Override
+							public synchronized Throwable fillInStackTrace() {
+								return this; // Make the output smaller by omitting the stacktrace
```

Comment:
Never saw this "blank stack trace" construct before.

Move the local class to a named class and extend ... `UnsupportedOperationException`? 🤔

Reply:
`UnwritableFileException`?

Reply:
I went with `UndeletableFileException`